### PR TITLE
Add release task

### LINKED ISSUE #594: Add a task for releasing
https://gitter.im/luckyframework/Lobby?at=5b880e01cff55e561781f175

We get this question a lot: "how do I make a release build for my app". It would be good to have a task that abstracts this.

### PR DESCRIPTION
**Pull request by HarrisonB**
Fixes #594 

Not exactly sure how to add tests for this (though I'm not sure if they're necessary).

### CODE REVIEW

**Comment by edwardloveall** [comment]
Awesome @HarrisonB Thanks so much for taking this on! It may take us a couple days to look at this but on first glance it looks like a great start.

File: spec/tasks/build/release_spec.cr [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  it "does not create executable if the build fails" do
+    with_cleanup do
+      Dir.mkdir_p("./src")
+      File.write "./src/server.cr", %({{ raise "this build will fail" }})
+      Build::Release.new.call
+      File.exists?("./server").should be_false
+    end
+  end
+end
```
Comment (paulcsmith) [inline spec/tasks/build/release_spec.cr]: Nice specs!

**Comment by paulcsmith** [comment]
Thanks for tackling this!

File: spec/tasks/build/release_spec.cr (hunk 1 above)
Comment (HarrisonB) [reply spec/tasks/build/release_spec.cr]: Thanks 😄

**Pull request merged**